neat_cache/test/redis_test.dart

```diff
@@ -0,0 +1,182 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'package:test/test.dart';
+import 'package:neat_cache/neat_cache.dart';
+import 'package:neat_cache/src/providers/redis.dart';
+import 'package:neat_cache/cache_provider.dart';
+import 'utils.dart';
+
+void _doAndForget(Function fn) {
+  scheduleMicrotask(() async {
+    try {
+      await fn();
+    } catch (_) {
+      // ignore
+    }
+  });
+}
+
+class SocketProxyServer {
+  ServerSocket _server;
+  final List<Socket> _sockets = [];
+  final String _targetHost;
+  final int _targetPort;
+
+  /// Number of accepted connections (including those closed).
+  int acceptedConnections = 0;
+
+  /// Delay before forwarding to the target.
+  Duration proxyDelay = Duration(milliseconds: 0);
+
+  SocketProxyServer._(this._server, this._targetHost, this._targetPort);
+
+  static Future<SocketProxyServer> create(
+    String targetHost,
+    int targetPort,
+  ) async {
+    final server = await ServerSocket.bind('localhost', 0);
+    final sp = SocketProxyServer._(server, targetHost, targetPort);
+    scheduleMicrotask(() async {
```

Comment:
this could be just `server.forEach(sp._process);` without the microtask wrapping, couldn't it? Or could server be a blocking mock impl?

Reply:
Server could be mocking, but mocking redis is a lot of work...

And yeah, I didn't need to have a  microtask here, I added to avoid the lint that I'm not awaiting all futures :)